pkg/controllers/v1alpha1/dataset/dataset_controller.go

```diff
@@ -119,6 +121,26 @@ func (r *DatasetReconciler) reconcileDataset(ctx reconcileRequestContext) (ctrl.
 		}
 	}
 
+	// 4. Scale the runtime controller to handle the phase change
+	if ctx.Dataset.Status.Phase == datav1alpha1.NotBoundDatasetPhase {
```

Comment:
Moreover, I think maybe we should check runitme existence and scale out no matter what the dataset status is. Considering such cases:
(1) user upgrades fluid with a side effect which sets runtime controller's `replicas` back to 0. At this time there may be no `NotBounded` datasets to trigger the scaling out.
(2) Another corner case might be that a Dataset is deleted but the bounded runtime is shutting down. If a Fluid upgrade happens at this time, the runtime will stuck at the shutting down phase.

Reply:
Thanks. It's fixed.